lib/cisco_node_utils/tacacs_server_group.rb

```diff
@@ -17,91 +18,134 @@
 # limitations under the License.
 
 require_relative 'node_util'
+require_relative 'tacacs_server'
 
 module Cisco
-  # TacacsServerGroup - node utility class for
-  # Raidus Server configuration management
+  # NXAPI implementation of AAA Server Group class
   class TacacsServerGroup < NodeUtil
     attr_reader :name
 
-    def initialize(name, instantiate=true)
-      unless name.is_a?(String)
-        fail ArgumentError, 'Invalid value (Name is not a String)'
-      end
-
+    def initialize(name, create=true)
+      fail TypeError unless name.is_a? String
       @name = name
 
-      create if instantiate
+      return unless create
+
+      TacacsServer.new.enable unless TacacsServer.enabled
+      config_set('tacacs_server_group', 'group', state: '', name: name)
     end
 
-    def self.tacacs_server_groups
-      hash = {}
-      group_list = config_get('tacacs_server_group', 'group')
-      return hash if group_list.nil?
+    def destroy
+      config_set('tacacs_server_group', 'group', state: 'no', name: @name)
+    end
 
-      group_list.each do |id|
-        hash[id] = TacacsServerGroup.new(id, false)
+    def servers
+      tacservers = config_get('tacacs_server_group', 'servers', @name)
+      servs = {}
+      unless tacservers.nil?
+        tacservers.each { |s| servs[s] = TacacsServerHost.new(s, false) }
```

Comment:
Confused... how does this work? 
You get a list of groups and then get a list of servers for each group; but it looks like the calls to `TacacsServerHost.new(s, false)` should just return right away because create is false...?